integrations/fastembed/src/haystack_integrations/components/embedders/fastembed/fastembed_sparse_document_embedder.py

```diff
@@ -0,0 +1,170 @@
+from typing import Any, Dict, List, Optional
+
+from haystack import Document, component, default_to_dict
+
+from .embedding_backend.fastembed_backend import _FastembedSparseEmbeddingBackendFactory
+
+
+@component
+class FastembedSparseDocumentEmbedder:
+    """
+    FastembedSparseDocumentEmbedder computes Document embeddings using Fastembed sparse models.
+
+    Usage example:
+    ```python
+    # To use this component, install the "fastembed-haystack" package.
+    # pip install fastembed-haystack
+
+    from haystack_integrations.components.embedders.fastembed import FastembedSparseDocumentEmbedder
+    from haystack.dataclasses import Document
+
+    doc_embedder = FastembedSparseDocumentEmbedder(
+        model="prithvida/Splade_PP_en_v1",
+        batch_size=256,
+    )
+
+    doc_embedder.warm_up()
+
+    # Text taken from PubMed QA Dataset (https://huggingface.co/datasets/pubmed_qa)
+    document_list = [
+        Document(
+            content="Oxidative stress generated within inflammatory joints can produce autoimmune phenomena and joint destruction. Radical species with oxidative activity, including reactive nitrogen species, represent mediators of inflammation and cartilage damage.",
+            meta={
+                "pubid": "25,445,628",
+                "long_answer": "yes",
+            },
+        ),
+        Document(
+            content="Plasma levels of pancreatic polypeptide (PP) rise upon food intake. Although other pancreatic islet hormones, such as insulin and glucagon, have been extensively investigated, PP secretion and actions are still poorly understood.",
+            meta={
+                "pubid": "25,445,712",
+                "long_answer": "yes",
+            },
+        ),
+    ]
+
+    result = doc_embedder.run(document_list)
+    print(f"Document Text: {result['documents'][0].content}")
+    print(f"Document Embedding: {result['documents'][0].sparse_embedding}")
+    print(f"Embedding Dimension: {len(result['documents'][0].sparse_embedding)}")
+    ```
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        model: str = "prithvida/Splade_PP_en_v1",
+        cache_dir: Optional[str] = None,
+        threads: Optional[int] = None,
+        prefix: str = "",
+        suffix: str = "",
+        batch_size: int = 256,
```

Comment:
@lambda-science as you observed, this makes memory explode. I would default to a lower number...

Reply:
Maybe put thread and batch size default to 1 ? I really have to idea what to put as default it's kind of hard choice

Reply:
@Anush008 do you have any suggestions? 256 makes Colab go OOM

Reply:
What if we just put batch_size = 1 and let the user decide if he wants a speed-up

Reply:
I too don't have a definitive number. Since sparse vectors are expensive we can start with a modest low number. 64 maybe.

Reply:
Maybe there have been a fix in 0.2.5 ? because batch_size=256 (only like 5go ram) doesn't make my computer explode anymore